source/dotnet/Library/AdaptiveCards/AdaptiveCardConverter.cs

```diff
@@ -20,29 +20,52 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
 
         public override bool CanWrite => false;
 
+        private void ValidateJsonVersion(ref JObject jObject)
+        {
+            string exceptionMessage = "";
+            if (jObject.Value<string>("version") == null)
+            {
+                exceptionMessage = "Could not parse required key: version. It was not found.";
+            }
+
+            // If this is the root AdaptiveCard and missing a version we fail parsing.
+            // The depth checks that cards within a Action.ShowCard don't require the version
+            if (jObject.Value<string>("version") == "")
+            {
+                exceptionMessage = "Property is required but was found empty: version";
+            }
+
+            if (!String.IsNullOrEmpty(exceptionMessage))
+            {
+                if (AdaptiveCard.OnDeserializingMissingVersion == null)
```

Comment:
We can potentially reorder code to avoid nested if here ... I'll try and give it a shot

Reply:
The purpose of `OnDeserializingMissingVersion` is to allow the caller to have a say in what happens when the `version` property is *missing or empty*. We shouldn't want the caller to override every caller's version...

Reply:
I see ... this is not quite right.

---
In reply to: [325833010](https://github.com/microsoft/AdaptiveCards/pull/3462#discussion_r325833010) [](ancestors = 325833010)